apis/core/v1alpha1/resource.go

```diff
@@ -265,3 +265,20 @@ type TargetSpec struct {
 type TargetStatus struct {
 	ConditionedStatus `json:",inline"`
 }
+
+// A Reference from one managed resource to another.
+type Reference struct {
+	// Name of the referenced managed resource.
+	Name string `json:"name"`
+}
+
+// A Selector for a Reference from one managed resource to another.
+type Selector struct {
+	// MatchController ensures that only managed resources with the same
+	// controller reference as the selecting resource will be selected.
+	MatchController *bool `json:"matchController,omitempty"`
```

Comment:
I really think `matchController` is a bad name since controller term is overloaded. `matchOwner`, `matchOwnerRef` or `matchControllerRef`?

Reply:
I agree that the term controller is overloaded, but in this case it is the term used by Kubernetes. `MatchControllerRef` sounds fine to me but an object can have multiple owners but only one controller so using `MatchOwner` or `MatchOwnerRef` does not feel indicative of actual behavior here.

Reply:
Yeah - `MatchControllerRef` SGTM. Less keen on `Owner` for the reasons @hasheddan pointed out.